File: app/src/services/bookingService.js
import api from '../config/api';

export const bookingService = {
  createBooking: async bookingData => {
    const response = await api.post('/bookings', bookingData);
    return response.data;
  },

  getMyBookings: async (status, page = 1) => {
    const response = await api.get('/bookings/my-bookings', {
      params: {status, page},
    });
    return response.data;
  },

  getSalonBookings: async (status, date, page = 1) => {
    const response = await api.get('/bookings/salon-bookings', {
      params: {status, date, page},
    });
    return response.data;
  },

  getBookingById: async bookingId => {
    const response = await api.get(`/bookings/${bookingId}`);
    return response.data;
  },

  updateBookingStatus: async (bookingId, status) => {
    const response = await api.put(`/bookings/${bookingId}/status`, {status});
    return response.data;
  },

  cancelBooking: async (bookingId, cancellationReason) => {
    const response = await api.put(`/bookings/${bookingId}/cancel`, {
      cancellationReason,
    });
    return response.data;
  },

  getAvailableTimeSlots: async (salonId, date, serviceIds) => {
    const response = await api.get('/timeslots/available', {
      params: {salonId, date, serviceIds: serviceIds.join(',')},
    });
    return response.data;
  },

  getSalonAvailableSlots: async (salonId, date) => {
    const response = await api.get(`/salons/${salonId}/available-slots`, {
      params: {date},
    });
    return response.data;
  },
};

File: luxtril_flutter/lib/services/booking_service.dart
import 'api_client.dart';
import '../models/booking.dart';

class BookingService {
  final ApiClient _api = ApiClient();

  Future<Booking> createBooking(Map<String, dynamic> data) async {
    final response = await _api.post('/bookings', body: data);
    return Booking.fromJson(response['booking'] ?? response);
  }

  Future<List<Booking>> getMyBookings({String? status}) async {
    final params = <String, String>{};
    if (status != null && status != 'all') params['status'] = status;
    final response = await _api.get('/bookings/my', queryParams: params.isNotEmpty ? params : null);
    final List<dynamic> data = response['bookings'] ?? response['data'] ?? [];
    return data.map((b) => Booking.fromJson(b)).toList();
  }

  Future<Booking> getBookingById(String id) async {
    final response = await _api.get('/bookings/$id');
    return Booking.fromJson(response['booking'] ?? response);
  }

  Future<Booking> cancelBooking(String id) async {
    final response = await _api.put('/bookings/$id/cancel');
    return Booking.fromJson(response['booking'] ?? response);
  }

  Future<List<Booking>> getSalonBookings(String salonId, {String? status}) async {
    final params = <String, String>{};
    if (status != null && status != 'all') params['status'] = status;
    final response = await _api.get('/bookings/salon/$salonId', queryParams: params.isNotEmpty ? params : null);
    final List<dynamic> data = response['bookings'] ?? response['data'] ?? [];
    return data.map((b) => Booking.fromJson(b)).toList();
  }

  Future<Booking> updateBookingStatus(String id, String status) async {
    final response = await _api.put('/bookings/$id/status', body: {'status': status});
    return Booking.fromJson(response['booking'] ?? response);
  }
}
